gusto/advection.py

```diff
@@ -73,17 +80,34 @@ def __init__(self, state, field, equation=None, *, solver_parameters=None,
         # equation class and we can get the correct function space from
         # the output function.
         if isinstance(equation, EmbeddedDGAdvection):
+            # check that the field and the equation are compatible
+            if not equation.V0.__eq__(field.function_space()):
```

Comment:
Can you not write `if equation.V0 != field.function_space()`?  If not, we should fix firedrake so you can.

Reply:
Ah yes I can, I'm not sure why I did it in this way!